pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -183,3 +181,36 @@ def __init__(self, set_variable, item) -> None:
         if not is_homogeneous:
             raise TypeError(f"Can't remove an element of type {item.dtype} from a set of {set_variable.dtype}")
         super().__init__(set_variable, item)
+
+class SetDiscard(SetMethod):
+    """
+    Represents a call to the .discard() method.
+
+    The discard() is a built-in method to remove elements from the set.
+    The discard() method takes exactly one argument. 
+    This method does not return any value.   
+
+    Parameters
+    ----------
+    set_variable : TypedAstNode
+        The name of the set.
+
+    item : TypedAstNode
+        The item to search for, and remove.
+    """
+    __slots__ = ()
+    _shape = None
+    _order = None
+    _rank = 0
+    _class_type = VoidType()
+    name = 'discard'
+
+    def __init__(self, set_variable, item) -> None:
+        expected_type = set_variable.class_type.element_type
+        is_homogeneous = (
+            expected_type == item.class_type and
+            set_variable.rank - 1 == item.rank
+        )
+        if not is_homogeneous:
+            raise TypeError("Expecting an argument of the same type as the elements of the set")
```

Comment:
I think you should add a test for the case of passing an argument that is not of the same type

Reply:
Fixed : db25af1